src/Uno.Foundation/Uno.Core.Extensions/Uno.Core.Extensions.Collections/CollectionExtensions.cs

```diff
@@ -0,0 +1,385 @@
+// ******************************************************************
+// Copyright � 2015-2018 nventive inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ******************************************************************
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Uno.Collections;
+using Uno.Disposables;
+
+namespace Uno.Extensions
+{
+	/// <summary>
+	/// Provides Extensions Methods for ICollection.
+	/// </summary>
+	public static class CollectionExtensions
+	{
+		/// <summary>
+		/// Adds a new item with the default constructor
+		/// </summary>
+		/// <typeparam name="T"></typeparam>
+		/// <param name="items"></param>
+		/// <returns></returns>
+		public static T AddNew<T>(this ICollection<T> items)
+			where T : new()
+		{
+			T item = new T();
+
+			items.Add(item);
+
+			return item;
+		}
+
+		/// <summary>
+		/// Adds the items of the specified collection to the end of the ICollection.
+		/// </summary>
+		/// <typeparam name="T">The type of the items.</typeparam>
+		/// <param name="collection">Collection in which to insert items.</param>
+		/// <param name="items">The items to add.</param>
+		public static void AddRange<T>(this ICollection<T> collection, IEnumerable<T> items)
+		{
+			items.ForEach(collection.Add);
```

Comment:
I think this may perform better if I understand https://github.com/dotnet/roslyn/issues/5835 correctly. Can help to see the decompiled source for both.

```suggestion
			items.ForEach(item => collection.Add(item));
```